src/components/SavedCryptos.tsx

```diff
@@ -0,0 +1,115 @@
+  const [showAll, setShowAll] = useState(false);
+
+  const cryptoQueries = useQueries({
+    queries: props.favCryptos.map((crypto) => ({
+      queryKey: ["crypto", crypto.id],
+      queryFn: () => fetchCryptoData(crypto.id),
+    })),
+  });
+
+  const firstFourCryptos = props.favCryptos.slice(0, 4);
+  const remainingCryptos = props.favCryptos.slice(4);
+
+  return (
+    <>
+      <h2>Saved Cryptos</h2>
+      <div className="saved-cryptos-container">
+        <div className="saved-cryptos-list">
+          {remainingCryptos.length > 0 && (
+            <button
+              onClick={() => setShowAll(!showAll)}
+              className={`show-more-btn-mobile ${
+                showAll ? "btn-visible" : "btn-invisible"
+              }`}
+            >
+              {showAll ? "Show Less" : "Show More"}
+            </button>
+          )}
+          {props.favCryptos.length === 0 ? (
+            <p>No saved cryptocurrencies</p>
+          ) : (
+            <>
+              {firstFourCryptos.map((crypto, index) => {
+                const query = cryptoQueries[index];
+                if (query.isLoading)
+                  return <p key={crypto.id}>Loading {crypto.name}...</p>;
+                if (query.isError)
+                  return <p key={crypto.id}>Failed to load {crypto.name}</p>;
+                return (
+                  query.data && (
+                    <CryptoCard
+                      key={crypto.id}
+                      coinId={crypto.id}
+                      coinName={crypto.name}
+                      usd={query.data.usd}
+                      eur={query.data.eur}
+                      gbp={query.data.gbp}
+                      isSaved={props.favCryptos.some((c) => c.id === crypto.id)}
+                      onToggleSave={props.onSelectCrypto}
+                    />
+                  )
+                );
+              })}
+
+              {showAll &&
+                remainingCryptos.map((crypto, index) => {
+                  const query = cryptoQueries[index + 4];
+                  if (!query) return null;
+                  if (query.isLoading)
+                    return <p key={crypto.id}>Loading {crypto.name}...</p>;
+                  if (query.isError)
+                    return <p key={crypto.id}>Failed to load {crypto.name}</p>;
+                  return (
+                    query.data && (
+                      <CryptoCard
+                        key={crypto.id}
+                        coinId={crypto.id}
+                        coinName={crypto.name}
+                        usd={query.data.usd}
+                        eur={query.data.eur}
+                        gbp={query.data.gbp}
+                        isSaved={props.favCryptos.some(
+                          (c) => c.id === crypto.id
+                        )}
+                        onToggleSave={props.onSelectCrypto}
+                      />
+                    )
+                  );
+                })}
```

Comment:
you should keep a state or memo to manipulate the `visibleCryptos` state and not repeat these blocks